File: binary-tree-maximum-path-sum/binary-tree-maximum-path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = Integer.MIN_VALUE;
    public int util(TreeNode root)
    {
        if(root == null) return 0;
        
        int l = util(root.left);
        int r = util(root.right);
        
        int nodeMax = Integer.max(Integer.max(root.val,root.val + l + r),
                          Integer.max(root.val + l,root.val + r));
        ans = Integer.max(ans,nodeMax);
        int singleSum = Integer.max(root.val,Integer.max(root.val + l,root.val + r));
        
        return singleSum;
        
    }
    
    public int maxPathSum(TreeNode root) {
        ans = Integer.MIN_VALUE;
        util(root);
        return ans;
    }
}

File: binary-tree-maximum-path-sum/binary-tree-maximum-path-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def _init_(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    ans = -9999999
    def maxPathSum(self, root: TreeNode) -> int:
        Solution.ans = -99999999
        Solution.util(root)
        return Solution.ans;
    def util(root):
        if root == None:
            return 0
        l = Solution.util(root.left)
        r = Solution.util(root.right)
        nodeMax = max(root.val,root.val+l+r,root.val + l,root.val + r)
        Solution.ans = max(Solution.ans,nodeMax)
        singleSum = max(root.val,root.val+l,root.val+r)
        return singleSum